### PR TITLE
feat: add blob.uploadfrom(inputstream)

### PR DESCRIPTION
**Pull request by dmitry-fa**
uploadFrom(InputStream) is implemented

### CODE REVIEW

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java [hunk 1]
```diff
@@ -276,8 +276,8 @@ public void uploadFrom(Path path, BlobWriteOption... options) {
   /**
    * Uploads the given file path to this blob using specified blob write options and the provided
    * chunk size. The default chunk size is 2*1024*1024. Larger chunk sizes may improve the upload
-   * performance, but require more memory, which can cause OutOfMemoryError or add significant
-   * garbage collection overhead. Chunk sizes less than 256*1024 are not allowed, they will be
+   * performance but require more memory. It could cause OutOfMemoryError or add significant garbage
+   * collection overhead. Chunk sizes which are less than 256*1024 are not allowed, they will be
```
Comment (rahulKQL) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java]:
```suggestion
   * collection overhead. Chunk sizes which are less than 256KB are not allowed, they will be
```

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java [hunk 2]
```diff
@@ -292,15 +292,46 @@ public void uploadFrom(Path path, int chunkSize, BlobWriteOption... options) {
...
+   * @param options blob write options
+   * @throws StorageException upon failure
+   */
+  public void uploadFrom(InputStream input, BlobWriteOption... options) {
+    uploadFrom(input, DEFAULT_CHUNK_SIZE, options);
+  }
+
+  /**
+   * Uploads the given content to this blob using specified blob write options and the provided
+   * chunk size. The default chunk size is 2*1024*1024. Larger chunk sizes may improve the upload
```
Comment (rahulKQL) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java]:
```suggestion
   * chunk size. The default chunk size is 2MB. Larger chunk sizes may improve the upload
```

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java [hunk 3]
```diff
@@ -292,15 +292,46 @@ public void uploadFrom(Path path, int chunkSize, BlobWriteOption... options) {
...
+   * performance but require more memory. It could cause OutOfMemoryError or add significant garbage
+   * collection overhead. Chunk sizes which are less than 256*1024 are not allowed, they will be
```
Comment (rahulKQL) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java]: Here as well

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java [hunk 4]
```diff
@@ -292,15 +292,46 @@ public void uploadFrom(Path path, int chunkSize, BlobWriteOption... options) {
     if (Files.isDirectory(path)) {
       throw new StorageException(0, path + ": Is a directory");
     }
+    try {
+      try (InputStream input = Files.newInputStream(path)) {
+        uploadFrom(input, chunkSize, options);
+      }
+    } catch (IOException e) {
+      throw new StorageException(e);
+    }
```
Comment (rahulKQL) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java]: Only one try block would be enough
```java
try(InputStream input = Files.newInputStream(path)) { 
  uploadFrom(input, chunkSize, options);
}  catch (IOException e) {
 throw new StorageException(e);
}
```

**Review by rahulKQL — CHANGES_REQUESTED** [review]

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java (hunk 1 above)
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java]: done

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java (hunk 2 above)
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java]: done

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java (hunk 4 above)
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java]: Sure :) Thanks for catching this!

**Review by rahulKQL — APPROVED** [review]

File: google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java (hunk 3 above)
Comment (dmitry-fa) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/Blob.java]: I think 2 MB is better than 2MB

**Pull request closed without merging**